llvm/lib/Target/AArch64/MCTargetDesc/AArch64MachObjectWriter.cpp

```diff
@@ -64,6 +64,10 @@ bool AArch64MachObjectWriter::getAArch64FixupKindMachOInfo(
   case FK_Data_2:
     Log2Size = Log2_32(2);
     return true;
+  case FK_PCRel_4:
+    Log2Size = Log2_32(4);
+    RelocType = unsigned(MachO::ARM64_RELOC_POINTER_TO_GOT);
```

Comment:
This doesn't look right. I'd expect it to follow similar logic to `FK_Data_4` and generate `ARM64_RELOC_UNSIGNED` by default [and `ARM64_RELOC_POINTER_TO_GOT` based on the symbol kind].

Reply:
(specifically, the 495e374f8fb370a0bf45304fbc835dbf78f2949d commit doesn't look right)

Reply:
These are experimental changes trying to move the needle a bit to get apps working. Here are the details for this one https://github.com/dotnet/runtime/issues/67232#issuecomment-1087985783.

If you want play with it, you can checkout my this PR branch in llvm-project, apply  https://github.com/dotnet/runtime/compare/main...am11:feature/nativeaot/osx-arm64 on runtime's `main`, and then use this script: https://github.com/dotnet/runtime/pull/68211#discussion_r854238425 to test the whole thing. Rinse and repeat.

Once you are there deep in lldb session, it would be easier to get the feel what kind of behavior we get by tweaking `FK_PCRel_4` case (spoiler alert; it ain't pretty). :grinning:

Reply:
I actually went through the issue and reproduced all the same failures as you did :)

Reply:
I might get back to it in a few days. Meanwhile, if you find the culprit, feel free to add more commits. I have added you as collaborator in my forks.